merge_duplicate_contacts/wizard/merge_contact.py

```diff
@@ -0,0 +1,935 @@
+import itertools
+import logging
+import operator
+from ast import literal_eval
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    def open_wizard_action(self):
+        context = {}
+        data = {
+            "group_by_name": True,
+            "state": "option",
+            "number_group": 0,
+            "current_line_id": False,
+            "line_ids": [],
+            "partner_ids": [],
+            "exclude_contact": True,
+            "maximum_group": 0,
+            "total_duplicates": 0,
+            "duplicate_position": 0,
+            "associate_contact": True,
+            "contact_not_being_customer": True,
+            "without_sales_orders": True,
+        }
+        wizard = self.env["base.partner.merge.automatic.wizard"].create(data)
+        wizard.with_context(context=context)._process_query(
+            "select min(id), array_agg(id) from res_partner where id in %s"
+            % (tuple(self.ids),),
+            ignore_occurence=False,
+        )
+        return wizard._action_new_next_screen()
+
+
+class MergePartnerLine(models.TransientModel):
+    _inherit = "base.partner.merge.line"
+    _order = "len_aggr_ids asc, min_id asc"
+
+    len_aggr_ids = fields.Integer(
```

Comment:
An explicit field name is better than shortcuts (even if Odoo does it).

Moreover, for a count, it's better to name it with count `aggr_ids_count`